### ISSUE TITLE
A little bit of feedback and some questions

### ISSUE THREAD

**Issue by dobrite** [kvark/froggy#4]
I toyed with Froggy last night and really enjoyed the small API. It felt vastly more natural than a traditional ECS system.

I have a few questions and some feedback:

1) I'm having trouble understanding what to do about "dead" components. It seems that iteration exposes these (as noted in a [comment](https://github.com/kvark/froggy/blob/master/src/lib.rs#L161)). For example, bullets that should be removed when they reach off screen. Is client code expected to only deal in `Pointer` types? Is there some API planned but not implemented?

2) I feel that `pin` should be named something more like `get_ptr` or `to_ptr` as it is getting a `Pointer` to a particular index, rather than storing a component at a particular index.

3) The library is now a month or two old. Any current thoughts on it? Just an experiment, or will it live on?

Finally I'd like to say thank you for all the work you put in. I enjoy your presence in the Rust community.

Cheers 🍻 
Dave

**Comment by vitvakatu** [comment]
Hello, @dobrite 
Actually, I can answer for only third question, I remembered the conversation with *kvark* about it.
Yep, for now, it's more an experiment than ready-to-use library. Unfortunately, *kvark* doesn't have enough time to work on it for now. (It seems he doesn't have time to answer your questions, actually :smile: )
But I can't say that `froggy` is abandoned. Perhaps, in future it will replace `specs`.
I want to play with `froggy` a bit, but I'm busy with work and study as well. Maybe we should work together to improve that lib?

**Comment by kvark** [comment]
@dobrite I messed up my Github notifications, sorry!

> I'm having trouble understanding what to do about "dead" components. It seems that iteration exposes these

Good question! We were just discussing it with @vitvakatu and concluded that we'll not expose the storages as slices. Thus, `Deref` implementations will be gone, and the same for `access_index`. You'd be able to produce a pointer from iterator, and we'll have iterators. Those will also walk over "dead" components.

Now, the "dead" components are not actually dead. They are just not references by any pointer, and our current memory model doesn't say that a component dies instantly. It can still hang out in the storage for some time, and you are able to "revive" it by pinning it with a pointer. That is currently by design.

> I feel that pin should be named something more like get_ptr or to_ptr

That's up to consideration still. Naming is hard! Let's revisit this after we remove `pin` and `access_index` ;)

> The library is now a month or two old. Any current thoughts on it? Just an experiment, or will it live on?

I've been quite busy since I rolled in the initial Froggy version and wasn't able to find time for some dog-fooding. Very happy to see the interest in it, and thanks to @vitvakatu it's already moving forward.

So far, Froggy hasn't lost even a bit of excitement on me. But I don't see it replacing Specs, but rather serving an alternative for those who find ECS too limiting and/or inconvenient.
